### PR TITLE
Add demos for lab08.

### PR DESCRIPTION
**Pull request by gabrielmocanu**
Add demos for lab08.

Signed-off-by: Gabriel Mocanu <gabi.mocanu98@gmail.com>

### CODE REVIEW

**Comment by gabrielmocanu** [comment]
@cristian-vijelie  Can you take a look on this, please?

File: laborator/content/stiva/demo/stack-addressing/Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+NASM = nasm
+AFILES = stack-addressing.asm 
+OBJS = $(AFILES:.asm=.o) 
+ASM_FLAGS = -f elf32 -g
+LD=gcc
+LDFLAGS = -m32 -g
```
Comment (cristian-vijelie) [inline laborator/content/stiva/demo/stack-addressing/Makefile:6]:
```suggestion
LDFLAGS = -m32 -g -no-pie
```

File: laborator/content/stiva/demo/stack-addressing/stack-addressing-2.asm [hunk 2]
```diff
@@ -0,0 +1,50 @@
+;
+; Author: Gabriel Mocanu <gabi.mocanu98@gmail.com>
+;
+
+%include "../utils/printf32.asm"
+
+%define NUM 5
```
Comment (cristian-vijelie) [inline laborator/content/stiva/demo/stack-addressing/stack-addressing-2.asm:7]:
```suggestion
```

Comment (cristian-vijelie) [reply laborator/content/stiva/demo/stack-addressing/stack-addressing-2.asm:7]: It's not used, remove it

File: laborator/content/stiva/demo/stack-addressing/stack-addressing-2.asm [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+   
+section .text
+
+; esp -> stack pointer
+; ebp -> base pointer
+
+extern printf
+global main
```
Comment (cristian-vijelie) [inline laborator/content/stiva/demo/stack-addressing/stack-addressing-2.asm:15]:
```suggestion
section .text
    extern printf
    global main

; esp -> stack pointer
; ebp -> base pointer
```

File: laborator/content/stiva/demo/stack-addressing/stack-addressing-2.asm [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    ; VARIANTA 2
+    add esp, 20 ; 4 * nr_pushuri
+
+    ; VARIANTA 3
+    mov esp, ebp
+    
+    ; sub esp <-> add esp -> folosim doar pentru alocare/dezalocare de memorie 
+
+
```
Comment (cristian-vijelie) [inline laborator/content/stiva/demo/stack-addressing/stack-addressing-2.asm:44]:
```suggestion
```

File: laborator/content/stiva/demo/stack-addressing/stack-addressing-2.asm [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+global main
+main:
+    mov ebp, esp
+    
+    push dword 10 ; sub esp, 4;  mov [esp], 10;
+    push dword 11 ; sub esp, 4;  mov [esp], 11;
+    push dword 12 ; sub esp, 4;  mov [esp], 12;
+    push dword 13 ; sub esp, 4;  mov [esp], 13;
+    push dword 14 ; sub esp, 4;  mov [esp], 13;
+
```
Comment (cristian-vijelie) [inline laborator/content/stiva/demo/stack-addressing/stack-addressing-2.asm:24]:
```suggestion
```

File: laborator/content/stiva/demo/stack-addressing/stack-addressing.asm [hunk 6]
```diff
@@ -0,0 +1,37 @@
+;
+; Author: Gabriel Mocanu <gabi.mocanu98@gmail.com>
+;
+
+%include "../../utils/printf32.asm"
+
+%define NUM 5
```
Comment (cristian-vijelie) [inline laborator/content/stiva/demo/stack-addressing/stack-addressing.asm:7]:
```suggestion
```

Comment (cristian-vijelie) [reply laborator/content/stiva/demo/stack-addressing/stack-addressing.asm:7]: Not used, remove it

File: laborator/content/stiva/demo/stack-addressing/stack-addressing.asm [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+   
+section .text
+
+extern printf
+global main
+main:
```
Comment (cristian-vijelie) [inline laborator/content/stiva/demo/stack-addressing/stack-addressing.asm:13]:
```suggestion
section .text
    extern printf
    global main

main:
```

File: laborator/content/stiva/demo/stack-addressing/stack-addressing.asm [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    mov ebp, esp
+    
+    push dword 10
+    push dword 11
+    push dword 12
+    push dword 13
+
```
Comment (cristian-vijelie) [inline laborator/content/stiva/demo/stack-addressing/stack-addressing.asm:20]:
```suggestion
```

**Review by cristian-vijelie — CHANGES_REQUESTED** [review]
I think that the Makefile should also build the 2nd demo, too.

**Comment by RazvanN7** [comment]
@gabrielmocanu Please address the review and let's get this in.

File: laborator/content/stiva/demo/sum_n_interm/sum_n_interm.asm [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+main:
+    push    ebp
+    mov     ebp, esp
+    mov     dword [i], 1
+.L3:
+    mov     eax, [i]
+    cmp     eax, 100
+    ja      .L2
+    add [sum], eax
```
Comment (cristian-vijelie) [inline laborator/content/stiva/demo/sum_n_interm/sum_n_interm.asm:19]: E intentionata diferenta de aici de coding-style? <instr><spatiu><argumente> vs <instr><tab><argumente>

**Review by cristian-vijelie — CHANGES_REQUESTED** [review]
Mai sunt cateva schimbari necesare. Am adaugat 2 comment-uri, care se aplica in mai multe locuri

**Pull request merged**